frontend/pages/dashboard/new-post.tsx

```diff
@@ -1,17 +1,63 @@
+import React from 'react'
 import { NextPage } from 'next'
+import { useRouter } from 'next/router'
+import { Node } from 'slate'
+import { withApollo } from '../../lib/apollo'
+
+import DashboardLayout from '../../components/Layouts/DashboardLayout'
 import JournalyEditor from '../../components/JournalyEditor'
+import { useCreatePostMutation } from '../../generated/graphql'
+
+const initialValue = [
+  {
+    type: 'paragraph',
+    children: [{ text: '' }],
+  },
+]
+
+const NewPostPage: NextPage = () => {
+  const [title, setTitle] = React.useState<string>('')
+  const [body, setBody] = React.useState<Node[]>(initialValue)
+
+  const router = useRouter()
+  const [createPost] = useCreatePostMutation({
+    onCompleted: (post) => {
+      router.push({
+        pathname: '/post/[id]',
+        query: { id: post.id },
+      })
+    },
+  })
+
+  const createNewPost = (e) => {
+    e.preventDefault()
+    createPost({
+      variables: { title, body },
+    })
+  }
 
-const NewPostPage: NextPage = () => (
-  <>
-    <h1>Let's write a post</h1>
-    <JournalyEditor />
-    <style jsx>{`
-      h1 {
-        margin: 50px auto;
-        text-align: center;
-      }
-    `}</style>
-  </>
-)
+  return (
+    <DashboardLayout>
```

Comment:
Wooohoooooooo so glad this is now going to be inside the dashboard haha I was just pondering doing that